RingPublishingGDPR/src/main/java/com/ringpublishing/gdpr/RingPublishingGDPR.java

```diff
@@ -284,42 +318,90 @@ private boolean isOutdated()
         return outdated;
     }
 
-    private void verifyConsents(@NonNull Context applicationContext)
+    private void verifyOutdated()
     {
         final Map<String, String> consents = storage.getRingConsents();
 
         if (consents == null || consents.isEmpty())
         {
             Log.w(TAG, "Fail verify consents. Consents are empty");
+            setVerifyState(VerifyState.FAILURE);
             return;
         }
 
-        api.verify(consents, new VerifyApiCallback()
+        api.verify(consents, new VerifyCallback()
         {
             @Override
             public void onOutdated(String rawStatus)
             {
-                storage.saveLastAPIConsentsCheckStatus(rawStatus);
                 storage.setOutdated(true);
-                FormView formView = getFormView(applicationContext);
-                showFromApplication(formView);
+                storage.saveLastAPIConsentsCheckStatus(rawStatus);
+                setVerifyState(VerifyState.OUTDATED);
             }
 
             @Override
             public void onActual(String rawStatus)
             {
-                storage.saveLastAPIConsentsCheckStatus(rawStatus);
                 storage.setOutdated(false);
+                storage.saveLastAPIConsentsCheckStatus(rawStatus);
+                setVerifyState(VerifyState.ACTUAL);
             }
 
             @Override
-            public void onFail(String status)
+            public void onFailure(String status)
             {
                 storage.saveLastAPIConsentsCheckStatus(status);
+                setVerifyState(VerifyState.FAILURE);
             }
         });
     }
 
+    private void setVerifyState(VerifyState verifyState)
+    {
+        this.verifyState = verifyState;
+        apiCallFinished();
+    }
+
+    private synchronized void apiCallFinished()
+    {
+        final TenantState tenantState = tenantConfiguration.getState();
+
+        if (tenantState == TenantState.LOADING || verifyState == VerifyState.LOADING)
```

Comment:
a nie możesz tego jakimś semaforem ograć na poziomie requestów do API zamiast wprowadzać stan LOADING?

Reply:
Semafory informują tylko o tym, czy jednym odpalić drugi lub czy któryś się skończył lub czy oba się skończyły.
Ja tu chciałem mieć informację od razu z jakim stanem się skończyły. 
Semafory służą głównie do operacji na wątkach, tutaj retrofit sam zarządza wątkami i w randomowym z nich daje nam odpowiedź, to semafor tu średnio pasuje.

Mógłbym pewnie tu coś komplikować ale tak jest czytelniej.
Przeniosę najwyżej ten jeden stan do głównej klasy i z tych 2 stanów zrobię jedną klaskę, może będzie bardziej odseparowane.

Reply:
dlatego napisałem "semafor", a nie semafor :) - Można dosłownie odwzorować iOS-owe DispatchGroups użyte przez Adama do tego samego celu. Może można inaczej. Natomiast stan Loading nie jest realnym stanem biznesowym i naprawdę fajnie by było nie mieszać tutaj 2 różnych systemów walutowych.